server/functions/framework/compiled_function.go

```diff
@@ -504,13 +501,18 @@ func polymorphicTypesCompatible(paramTypes []pgtypes.DoltgresType, exprTypes []p
 	var baseType pgtypes.DoltgresType
 	for i, paramType := range paramTypes {
 		if polymorphicParamType, ok := paramType.(pgtypes.DoltgresPolymorphicType); ok {
+			baseExprType := exprTypes[i]
 			// Although we do this check before we ever reach this function, we do it again as we may convert anyelement
 			// to anynonarray, which changes type validity
-			if !polymorphicParamType.IsValid(exprTypes[i]) {
+			// Unknown type is valid for any type
+			_, isUnknown := baseExprType.(pgtypes.UnknownType)
+			if isUnknown {
+				continue
+			}
```

Comment:
We should change the `ok` check above to `ok && exprTypes[i].BaseID() != pgtypes.DoltgresTypeBaseID_Unknown`. Then delete this portion.